### PR TITLE
Add read by name workaround

### PR DESCRIPTION
**Pull request by JaurbanRH**
Some entities doesn't have properly set entity_e hencenam read_by_name doesn't work on all entites.
@jsmadis @mganisin  please review

### CODE REVIEW

**Comment by mganisin** [comment]
@JaurbanRH would you mind contribute also some test(s). Even trivial would be nice

**Comment by mkudlej** [comment]
@jaurbanRH would you mind to provide some examples of objects without names?

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -423,3 +423,17 @@ def provider_account(provider_account_params, api):
...
+
+@pytest.fixture(scope='module')
+def account_plans_params():
+    suffix = get_suffix()
+    name = f"test-{suffix}"
+    return dict(name=name)
+
+
+@pytest.fixture(scope='module')
+def account_plan(account_plans_params, api):
```
Comment (mkudlej) [inline tests/integration/conftest.py:436]: I think in this case is better to fix account plan
`class AccountPlan(DefaultResource):
    def __init__(self, entity_name='name', **kwargs):  <--- system_name to name
        super().__init__(entity_name=entity_name, **kwargs)`

File: threescale_api/defaults.py [hunk 2]
```diff
@@ -178,6 +178,8 @@ def read_by_name(self, name: str, **kwargs) -> 'DefaultResource':
         for item in self._list(**kwargs):
             if item.entity_name and item.entity_name == name:
                 return item
+            if item.get("name") == name:
```
Comment (mkudlej) [inline threescale_api/defaults.py:181]: I know in description of function is that it searches in system_name, name and org_name, but I don't recommend to change this and I prefer to fix account plan definition.

**Review by mkudlej — CHANGES_REQUESTED** [review]

File: tests/integration/conftest.py (hunk 1 above)
Comment (mkudlej) [reply tests/integration/conftest.py:436]: @jsmadis @mganisin What do you think about my proposal?

**Review by mkudlej — APPROVED** [review]
LGTM

**Review by jsmadis — APPROVED** [review]

**Pull request merged**